sds_data_manager/lambda_code/SDSCode/pipeline_lambdas/batch_starter.py

```diff
@@ -525,17 +526,142 @@ def generate_queue_url(event):
     return queue_url
 
 
+def get_latest_repoint_file_from_db(session):
+    """Retrieve the latest version of repoint file.
+
+    Parameters
+    ----------
+    session : sqlalchemy.orm.Session
+        Active SQLAlchemy session for querying the database.
+
+    Returns
+    -------
+    str
+        The file name for the latest repoint file.
+    """
+    # Look for latest and greatest file_path in the repointing table.
+    query = (
+        session.query(models.RepointTable)
+        .order_by(desc(models.RepointTable.file_path))
+        .limit(1)
+    )
+    latest_repoint_file = query.first()
+    if not latest_repoint_file:
+        raise ValueError("No repoint file found in the database.")
+
+    latest_repoint_file = latest_repoint_file.file_path
+    logger.info(f"Latest repoint file found: {latest_repoint_file}")
+    return latest_repoint_file
+
+
+def calculate_repoint_date_range(session, file_obj):
```

Comment:
This almost seems like we should just make another DB Table to store the items you need. Use this function to update the table with these start/end dates for all pointings in the file via upserts whenever we receive a repointing file. Then query that table directly for the information you need.

I'm still torn on which direction to go, but this seems like a decently complex version to do every time an ENA instrument gets kicked off in batch starter where you wouldn't want to download, open up files etc. unnecessarily.